src/server/scripts/EasternKingdoms/zone_elwynn_forest.cpp

```diff
@@ -414,11 +415,503 @@ class spell_quest_extincteur : public SpellScriptLoader
+{
+public:
+    npc_hogger() : CreatureScript("npc_hogger") { }
+
+    struct npc_hoggerAI : public ScriptedAI
+    {
+        npc_hoggerAI(Creature* creature) : ScriptedAI(creature)
+        {
+        }
+
+        void Reset() override
+        {
+            _events.Reset();
+
+            _minionsSummoned = false;
+            _endingSceneActive = false;
+            _alreadyEaten = false;
+            _isEating = false;
+
+            me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
+            me->SetReactState(REACT_AGGRESSIVE);
+            me->SetWalk(false);
+        }
+
+        void DamageTaken(Unit* attacker, uint32& damage) override
+        {
+            if (_endingSceneActive)
+            {
+                damage = 0;
+                return;
+            }
+
+            if (!_minionsSummoned && me->HealthBelowPctDamaged(50, damage))
+                SummonMinions();
+
+            if (!_alreadyEaten && me->HealthBelowPctDamaged(30, damage))
+                MoveToEatingPosition();
+
+
+            if (me->GetHealth() <= damage)
+            {
+                damage = 0;
+                me->SetHealth(1);
+                StartEndingScene();
+                RewardPlayers(attacker);
+            }
+
+            if (_isEating)
+            {
+                _isEating = false;
+                me->SetReactState(REACT_AGGRESSIVE);
+                Talk(SAY_STUNNED);
+            }
+        }
+
+        void EnterCombat(Unit* /*who*/) override
+        {
+            _events.ScheduleEvent(EVENT_VICIOUS_SLICE, 3000);
+        }
+
+        void MoveToEatingPosition()
+        {
+            _alreadyEaten = true;
+
+            if (Creature* target = me->FindNearestCreature(NPC_EATING_TARGET, 100.0f))
+            {
+                me->SetReactState(REACT_PASSIVE);
+                me->GetMotionMaster()->MovePoint(0, target->GetPositionX(), target->GetPositionY(), target->GetPositionZ(), true);
+                _events.ScheduleEvent(EVENT_CHECK_EAT_RANGE, 200);
+            }
+        }
+
+        void StartEndingScene()
+        {
+            _endingSceneActive = true;
+            _events.Reset();
+
+            me->SetReactState(REACT_PASSIVE);
+            me->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_IMMUNE_TO_PC);
+            me->StopMoving();
+            me->AttackStop();
+
+            Talk(SAY_BEG);
+
+            SummonGeneralHammondClay();
+            SummonAndromath();
+            SummonDumas();
+
+            _events.ScheduleEvent(EVENT_HAMMOND_GROUP_START_WALKING, 1000);
+        }
+
+        void RewardPlayers(Unit* killer)
+        {
+            if (Player* player = killer->GetCharmerOrOwnerPlayerOrPlayerItself())
+                player->RewardPlayerAndGroupAtEvent(NPC_HOGGER, me);
+        }
+
+        void SummonGeneralHammondClay()
+        {
+            TempSummon* hammond = me->SummonCreature(NPC_GENERAL_HAMMOND_CLAY, generalHammondClayPositions[0]);
+            if (hammond)
```

Comment:
You can merge those lines

if (TempSummon* hammond = me->SummonCreature(NPC_GENERAL_HAMMOND_CLAY, generalHammondClayPositions[0]))

Same for the next ifs